akka-docs/src/main/paradox/stream/operators/Source-or-Flow/collect.md

```diff
@@ -15,8 +15,42 @@ Apply a partial function to each incoming element, if the partial function is de
 ## Description
 
 Apply a partial function to each incoming element, if the partial function is defined for a value the returned
-value is passed downstream. Can often replace `filter` followed by `map` to achieve the same in one single operators.
+value is passed downstream. Can often replace `filter` followed by `map` to achieve the same in one single operator.
 
+@java[`collect` is supposed to be used with @apidoc[akka.japi.pf.PFBuilder] to construct the partial function.
+There is also a @ref:[collectType](collectType.md) that often can be easier to use than the `PFBuilder` and
+then combine with ordinary `filter` and `map` operators.]
+
+## Example
+
+Given stream element classes `Message`, `Ping`, and `Pong`, where `Ping` extends `Message` and `Pong` is an
+unrelated class. Ping and Pong have an int id field.
```

Comment:
This sentence structure is confusing. It would make sense if just introducing the snippet, maybe noting the int isn't needed.
```
Given stream element classes `Message`, `Ping`, and `Pong`, where `Ping` extends `Message` and `Pong` is an
unrelated class:
```

Reply:
that's right, let's remove that last part